test/utils/ConditionalOrderSignature.sol

```diff
@@ -46,7 +53,7 @@ contract ConditionalOrderSignature {
                 co.requireVerified,
                 co.trustedExecutor,
                 co.maxExecutorFee,
-                co.conditions
+                keccak256(abi.encode(hashedConditions))
```

Comment:
**Suggestion:** My main concern here is that the functionality is not really being tested, because the code in the contract is duplicated in the tests. This is a rare situation where I think a hardhat test is really worthwhile.

Reply:
really don't want to add hardhat to the project tbh but might be worth

Reply:
as much of a slog it probably is, it's probably worth it so this issue doesn't accidentally crop up again

Reply:
a lot of engineering hours were lost over this one, so it's definitely best to make sure it can't happen again

Reply:
Yeah I honestly don't think it is acceptable to deploy this code without it, because ultimately in its current state it is not tested. But totally understand the resistance to adding hardhat, it's a pretty ugly addition to the code base and a pain.

Reply:
Here is a reference test I wrote for exactly the same purpose before:
```ts
import { time, loadFixture } from "@nomicfoundation/hardhat-network-helpers";
import { anyValue } from "@nomicfoundation/hardhat-chai-matchers/withArgs";
import { expect } from "chai";
import { ethers } from "hardhat";

const SECONDS_IN_DAY: number = 60 * 60 * 24;
const SECONDS_IN_WEEK: number = SECONDS_IN_DAY * 7;

// Bump the timestamp by a specific amount of seconds
const timeTravel = async (seconds: number): Promise<number> => {
  return time.increase(seconds);
};

describe("CollectorDAO", function () {
  // We define a fixture to reuse the same setup in every test.
  // We use loadFixture to run this setup once, snapshot that state,
  // and reset Hardhat Network to that snapshot in every test.
  async function deployOneYearLockFixture() {
    // Contracts are deployed using the first signer/account by default
    const [owner, otherAccount] = await ethers.getSigners();

    const CollectorDAO = await ethers.getContractFactory("CollectorDAO");
    const collectorDAO = await CollectorDAO.deploy();

    return { collectorDAO, owner, otherAccount };
  }

  describe("Signing votes", function () {
    it("Can successfully submit and sign a vote", async function () {
      const { collectorDAO, owner } = await loadFixture(
        deployOneYearLockFixture
      );

      await collectorDAO.buyMembership({ value: ethers.utils.parseEther("1") });

      const targets = [owner.address];
      const values = [1000];
      const signatures = ["0x00"];

      await collectorDAO.propose(targets, values, signatures);

      const proposalId = await collectorDAO.getProposalId(
        0,
        targets,
        values,
        signatures
      );
      const voteFor = true;

      const domain = {
        name: "CollectorDAO",
        version: "1",
        chainId: 31337,
        verifyingContract: collectorDAO.address,
      };

      const types = {
        Ballot: [
          { name: "proposalId", type: "bytes32" },
          { name: "voteFor", type: "bool" },
        ],
      };

      const ballot = {
        proposalId,
        voteFor,
      };

      const signature = await owner._signTypedData(domain, types, ballot);

      const expectedSignerAddress = owner.address;
      const recoveredAddress = ethers.utils.verifyTypedData(
        domain,
        types,
        ballot,
        signature
      );

      expect(recoveredAddress).to.equal(expectedSignerAddress);

      const splitSignature = ethers.utils.splitSignature(signature);

      const { v, r, s } = splitSignature;

      await collectorDAO.submitSignedVote(ballot, v, r, s);

      await timeTravel(SECONDS_IN_WEEK * 2);

      await collectorDAO.execute(proposalId, targets, values, signatures);

      const proposal = await collectorDAO.proposals(proposalId);

      expect(proposal.executed).to.be.true;
      expect(proposal.forVotes).to.equal(1);
    });
  });
});
```